Makefile

```diff
@@ -83,55 +86,78 @@ bin:         ## Build all go binaries from cmd/.
 else
 	# TODO(TheSpiritXIII): Temporary env variables part of `export.go` unit tests.
 	$(call docker_build, -f ./hack/Dockerfile --target sync -o . -t gmp/hermetic \
 		--build-arg RUNCMD='GIT_TAG="$(shell git describe --tags --abbrev=0)" TEST_TAG=true ./hack/presubmit.sh test' .)
 	rm -rf vendor.tmp
 endif
 
-kindtest:    ## Run e2e test suite against fresh kind k8s cluster.
+.PHONY: e2e
+e2e:         ## Run e2e test suite against fresh kind k8s cluster. By default it
+             ## deploys and uses single-node kind cluster (left up after the tests). Setting
+             ## NO_DOCKER=1 execute tests from the host machine assuming kubectl
+             ## is already configured against the Kubernetes cluster of you choice.
              ##
-	@echo ">> building image"
+             ## Setting NO_DOCKER=1 also writes real data to GCM API under $PROJECT_ID
+             # environment variable. Use $GMP_CLUSTER, $GMP_LOCATION to specify
+             ## timeseries labels.
+             ##
+ifeq ($(NO_DOCKER), 1)
+	@echo ">> running e2e against your cluster"
+	go test `go list ./... | grep e2e` -args -project-id=${PROJECT_ID} -cluster=${GMP_CLUSTER} -location=${GMP_LOCATION}
+else
+	@echo ">> building kindtest image"
 	$(call docker_build, -f hack/Dockerfile --target kindtest -t gmp/kindtest .)
 	@echo ">> running container"
 # We lose some isolation by sharing the host network with the kind containers.
 # However, we avoid a gcloud-shell "Dockerception" and save on build times.
 	docker run --network host --rm -v $(DOCKER_VOLUME):/var/run/docker.sock gmp/kindtest ./hack/kind-test.sh
+endif
 
-presubmit:   ## Run all checks and tests before submitting a change 
-             ## Use DRY_RUN=1 to only validate without regenerating changes.
+.PHONY: presubmit
+presubmit:   ## Regenerate all resources, build all images and run all tests.
+             ## Steps from presubmit are validated on the CI, but feel free to
+             ## run it if you see CI failures related to regenerating resources
+             ## or when you want to do local check before submitting.
+             ##
+             ## Use `CHECK=1` to fail the command if repo state is not clean
+             ## after presubmit (might require committing the changes).
              ##
-presubmit: updateversions regen bin test kindtest
+presubmit: updateversions regen bin test e2e
 
+.PHONY: updateversions
 CURRENT_TAG = v0.7.1-gke.0
 CURRENT_PROM_TAG = v2.35.0-gmp.5-gke.0
 CURRENT_AM_TAG = v0.24.0-gmp.3-gke.1
 LABEL_API_VERSION = 0.7.1
 FILES_TO_UPDATE = $(shell find . -type f -name "*.yaml" ! -name "kube-state-metrics.yaml" ! -name "node-exporter.yaml")
+updateversions: ## Modifie all manifests, so it contains the expected versions.
+                ##
+                ## TODO(bwplotka): CI does not check updateversions--add that there.
+                ## Also, consider moving updateversion to hack/presubmit.sh for
+                ## consistency.
```

Comment:
It would probably be better to use Kustomize for this. At some point I'll clean up PR #376 and add images into the Kustomization file!